### DISCUSSION TITLE
How to round properly in importers?

### DISCUSSION THREAD

**Discussion by nolith** [beancount/beancount#803] [General] [answered]
Hello, I'm writing my first importer and I have to perform some arithmetics and rounding

I'm confused by `beancount.core.number.round_to`, I was expecting it to round the number but it seems to truncate instead.

Here is a `pytest` showing my issue.

```python
======================================================== FAILURES ========================================================
_____________________________________________________ test_round_to ______________________________________________________

    def test_round_to():
      n = D('0.929627')

      r1 = round_to(n, D('0.1'))
      r2 = round_to(n, D('0.01'))

      assert r1 == D('0.9')
>     assert r2 == D('0.93')
E     AssertionError: assert Decimal('0.92') == Decimal('0.93')
E      +  where Decimal('0.93') = D('0.93')
```

Is there any other function I can use to get rounding on Decimal?

Thank you

**Comment by blais** [comment, marked as answer]
Use Decimal.quantize

**Reply by nolith** [reply]
Thanks, it worked great

**Comment by dnicolodi** [comment]
`round()` works just fine on `decimal.Decimal()` objects.

**Reply by nolith** [reply]
Thanks, it worked great

**Comment by dnicolodi** [comment]
By the way, `beancount.core.number.round_to()` works as expected and as documented. It is supposed to round down, not to round to nearest.

**Reply by nolith** [reply]
Thanks for pointing out that I overlooked the documentation of the function.

It is indeed correctly documented.

**Comment by nolith** [comment]
Thank you all for the help, I did a bit of extra investigation (I'm not a python developer) and for taxes in Italy the correct method is `Decimal.quantize` because it rounds the even up as requested by out tax agency.
